mapeditor/windownewmap.cpp

```diff
@@ -217,6 +221,18 @@ std::unique_ptr<CMap> generateEmptyMap(CMapGenOptions & options)
 	return map;
 }
 
+int getSelectedMapSize(QComboBox* comboBox, int dimension) {
+	QString selectedText = comboBox->currentText(); // Get the selected text
+	QRegularExpression regex(R"(\((\d+)x(\d+)\))"); // Regex for format (width x height)
```

Comment:
that's a questionable approach. I believe all available combobox options have format WxH, so you could avoid regex and simply split on `x`:
```cpp
return selectedText.split(QLatin1Char{'x'})[dimension - 1].toInt();
```
dimension being 1 or 2 is also a strange choice: you could return `QSize` object and then read whatever component you need. As I see, you use both components immediately, so returning QSize would be much better - making a single call, and you won't need the `dimension` param.
```cpp
const auto components = selectedText.split(QLatin1Char{'x'});
return QSize{components[0].toInt(), components[1].toInt()};
```

Reply:
Combo text is not simply "36x36". It's "S (36x36)". So it's can't be simply splited by "x".
![image](https://github.com/user-attachments/assets/3882af35-f320-4833-9e8f-4fcc3da763e1).

Reply:
I don't think there si better approach to get that size from text.

Reply:
Potential problem - if somebody were to translate these text, this could easily break. For example by using cyrillic 'Х' symbol in place of latin 'X'

Better solutions:
a) Fill combobox in code, on UI initialization AND set user data to actual size, e.g:
```
ui->sizeCombo->addItem(tr("S (36x36)"), QVariant(36));
```
And then retrieve said index, via `itemData` if I remember correctly.

b) Since combobox elements are known in advance, add fixed mapping for item index -> map size, e.g:
```cpp
static constexpr std::array indexToMapSize = { 36, 72, 108 ... }
int index = ui->sizeCombo->selectedIndex();
int mapSize = indexToMapSize[index];
```

Reply:
Got different idea right now. I forgot about that mapSizes vs Combo index.

std::pair<int, int> getSelectedMapSize(QComboBox* comboBox, const std::map<int, std::pair<int, int>>& mapSizes) {
	int selectedIndex = comboBox->currentIndex();

	auto it = mapSizes.find(selectedIndex);
	if (it != mapSizes.end()) {
		return it->second; // Return the width and height pair
	}

	return { 0, 0 };
}

Reply:
yeah, now it's way better